specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,328 @@
+    m_webview.GetDevToolsProtocolEventReceiver("Target.detachedFromTarget").DevToolsProtocolEventReceived += CoreWebView2_DetachedFromTarget;
+    m_webview.GetDevToolsProtocolEventReceiver("Target.targetInfoChanged").DevToolsProtocolEventReceived += CoreWebView2_TargetInfoChanged;
+    // Enable Runtime events for the default target of top page to receive Runtime.consoleAPICalled events, which is fired when console.log is called.
+    _ = m_webview.CallDevToolsProtocolMethodAsync("Runtime.enable", "{}");
+    // Auto attach to iframe and dedicated worker targets created from the default target of top page.
+    _ = m_webview.CallDevToolsProtocolMethodAsync("Target.setAutoAttach",
+        @"{""autoAttach"":true,""waitForDebuggerOnStart"":false,""flatten"":true}");
+  }
+```
+
+# API Details
+## Win32 C++
+```
+interface ICoreWebView2_10 : IUnknown {
+  /// Runs an asynchronous `DevToolsProtocol` method for a specific session of
+  /// an attached target.
+  /// There could be multiple `DevToolsProtocol` targets in a WebView.
+  /// Besides the top level page, iframes from different origin and web workers
+  /// are also separate targets. Attaching to these targets allows interaction with them.
+  /// These attachments to the targets are sessions and are identified by sessionId.
+  /// To use this API, you should set `flatten` parameter to true when calling
+  /// `Target.attachToTarget` or `Target.setAutoAttach` `DevToolsProtocol` method.
+  /// Using `Target.setAutoAttach` is recommended as that would allow you to attach
+  /// to dedicated worker target, which is not discoverable via other APIs like
+  /// `Target.getTargets`.
+  /// For more information about targets and sessions, navigate to
+  /// \[DevTools Protocol Viewer\]\[GithubChromedevtoolsDevtoolsProtocolTotTarget\].
+  /// For more information about available methods, navigate to
+  /// \[DevTools Protocol Viewer\]\[GithubChromedevtoolsDevtoolsProtocolTot\]
+  /// The `sessionId` parameter is the sessionId for an attached target.
+  /// nullptr or empty string is treated as the session for the default target for the top page.
+  /// The `methodName` parameter is the full name of the method in the
+  /// `{domain}.{method}` format.  The `parametersAsJson` parameter is a JSON
+  /// formatted string containing the parameters for the corresponding method.
+  /// The `Invoke` method of the `handler` is run when the method
+  /// asynchronously completes.  `Invoke` is run with the return object of the
+  /// method as a JSON string.
+  ///
+  /// \[GithubChromedevtoolsDevtoolsProtocolTot\]: https://chromedevtools.github.io/devtools-protocol/tot "latest (tip-of-tree) protocol - Chrome DevTools Protocol | GitHub"
+  /// \[GithubChromedevtoolsDevtoolsProtocolTotTarget\]: https://chromedevtools.github.io/devtools-protocol/tot/Target "Chrome DevTools Protocol - Target domain"
+
+  HRESULT CallDevToolsProtocolMethodForSession(
+      [in] LPCWSTR sessionId,
```

Comment:
We should move the session ID parameter to be the last parameter to match the .NET/WinRT signature

Reply:
+1. The motivation is that method overloads add parameters rather than rearrange them.

Reply:
No longer applicable given above.
Leave as is.